lib/bson/objectid.js

```diff
@@ -27,7 +27,8 @@ var checkForHexRegExp = new RegExp("^[0-9a-fA-F]{24}$");
 */
 var ObjectID = function ObjectID(id) {
   if(!(this instanceof ObjectID)) return new ObjectID(id);
-  if((id instanceof ObjectID)) return id;
+  // Duck-typing to support ObjectId from different npm packages
+  if((id instanceof ObjectID) || (id && id.toHexString)) return id;
```

Comment:
Highly questionable IMO, if you're going to ducktype I'd recommend using a more robust check. With this PR, a [tinycolor](https://github.com/bgrins/TinyColor) would be a valid ObjectId...